### PR TITLE
[Cleanup] Make Ru'Aun Gardens zone shift-proof

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

- Removes all tables from IDs file
- Use `GetFirstID()` with all npcs and mobs in the file.
- Correct portal event being wrongly assigned. The end-result is the same though, since that event does exactly the same as the new one, but now its retail-accurate.
- Document properly all regions.
- Accidentally added a text ID to full moon fountain, but thats gonna be needed anyway.

## Steps to test these changes

No changes should be observed.

### CODE REVIEW

File: scripts/zones/RuAun_Gardens/Zone.lua [hunk 1]
```diff
@@ -5,10 +5,78 @@ local ID = zones[xi.zone.RUAUN_GARDENS]
...
-    for k, v in pairs(ID.npc.PORTALS) do
-        zone:registerTriggerArea(k, unpack(v['coords']))
+local function handleClosingPortal(player, eventId, npcId)
+    if GetNPCByID(npcId):getAnimation() == xi.anim.OPEN_DOOR then
+        player:startOptionalCutscene(eventId)
+    end
+end
+
+local function handleGreenPortal(player)
+    if player:getCharVar('skyShortcut') == 1 then
```
Comment (TeoTwawki) [inline scripts/zones/RuAun_Gardens/Zone.lua:15]: later (not today) we should move this to the unlocks table, like so many other one off forever vars we've yet to deal with.

Comment (TeoTwawki) [reply scripts/zones/RuAun_Gardens/Zone.lua:15]: no wait I forgot, you can actually -lose- this one after gaining it, never mind! I forgot it breaks.

**Review by TeoTwawki — APPROVED** [review]

**Comment by TeoTwawki** [comment]
CI griping about newlines in zone file, but looks good aside from Ci issues

(its the switch cases I think)

File: scripts/zones/RuAun_Gardens/mobs/Groundskeeper.lua [hunk 2]
```diff
@@ -7,6 +7,26 @@ local ID = zones[xi.zone.RUAUN_GARDENS]
 -----------------------------------
 local entity = {}
 
+local despotPHTable =
+{
+    [ID.mob.DESPOT - 16] = ID.mob.DESPOT,
```
Comment (Frankie-hz) [inline scripts/zones/RuAun_Gardens/mobs/Groundskeeper.lua:12]: Edit: nvm

**Pull request merged**